### ISSUE TITLE
[BUG] toHaveUrl() does not take baseUrl into account

### ISSUE THREAD

**Issue by cyrilletuzi** [microsoft/playwright#8360]
**Context:**
- Playwright Version: 1.14
- Operating System: Mac
- Node.js version: 16.6
- Browser: All

**Code Snippet**

With this config:

```ts
import { PlaywrightTestConfig } from '@playwright/test';

const config: PlaywrightTestConfig = {
  webServer: {
    command: 'npm start',
    port: 4200,
  },
};

export default config;
```

and this code:

```ts
import { test, expect } from '@playwright/test';

test.describe('Account', () => {

  test('should be able to register successfully', async ({ page }) => {

    await page.goto('/account/register');

    // Do some stuff that will redirect to another page

    // Works
    await page.waitForURL('http://localhost:4200/account/login');
    await page.waitForURL('/account/login');
    await expect(page).toHaveURL('http://localhost:4200/account/login');

    // Does NOT work
    await expect(page).toHaveURL('/account/login');

  });

});
```

**Describe the bug**

The new 1.14 `toHaveUrl()` method seems to not take `baseUrl` into account.

**Comment by unnati914** [comment]
could i work on this?

**Comment by mxschmitt** [comment]
sure @unnati914!

**Comment by kevin-brotcke** [comment]
As a temporary workaround you can do something like:

```
await expect(this.page).toHaveURL(`${process.env.PLAYWRIGHT_TEST_BASE_URL}/${path}`);
```

**Comment by cyrilletuzi** [comment]
@mxschmitt Thanks for the fix. The issue is marked with v1.15, but the issue is still here in just released v1.15. I also tried with `next` version and issue is still here too. Is it normal?
